File: Spring/Chapter7/MyBatisSpringJavaInit/src/main/java/pers/init/dao/UserDao.java
package pers.init.dao;

import pers.init.entity.User;
import pers.init.pojo.UserPOJO;

import java.util.List;
import java.util.Map;

public interface UserDao {
    List<User> selectAll();
    User selectById(Integer id);
    User selectByMap(Map<String,String> map);
    User selectByPOJO(UserPOJO pojo);
    int insertUser1(User user);
    int insertUser2(User user);
    int updateUser(User user);
    int deleteUser(Integer id);
    List<User> selectBySqlColumn();
    List<Map<String,Object>> selectReturnMap();
    List<UserPOJO> selectReturnPOJO();
}

File: Spring/Chapter7/MyBatisSpringKotlinInit/src/main/kotlin/pers/init/dao/UserDao.kt
package pers.init.dao

import pers.init.entity.User
import pers.init.pojo.UserPOJO

interface UserDao {
    fun selectAll():List<User>
    fun selectById(id:Int):User
    fun selectByMap(map:Map<String,String>):User
    fun selectByPOJO(pojo: UserPOJO)
    fun insertUser1(user:User):Int
    fun insertUser2(user:User):Int
    fun updateUser(user:User):Int
    fun deleteUser(id:Int):Int
    fun selectBySqlColumn():List<User>
    fun selectReturnMap():List<Map<String,Any>>
    fun selectReturnPOJO():List<UserPOJO>
}
